planner/core/exhaust_physical_plans.go

```diff
@@ -2262,6 +2269,41 @@ func (la *LogicalAggregation) getHashAggs(prop *property.PhysicalProperty) []Phy
 		agg.SetSchema(la.schema.Clone())
 		hashAggs = append(hashAggs, agg)
 	}
+	// TODO: We haven't supported the agg algo with repartition.
```

Comment:
This prolonged code should be refined: here are some suggestions:
At first consider the required property: what if we cannot suffice a partitioned requirements? we should return nil, otherwise, we should **refuse** it.
Then if the requirements is matched, we have three cases: one-phase, two-phase totally in TiFlash, two-phase partly in TiFlash and partly in TiDB without an extra shuffle. Currently, a scalar agg should only belog to the third scenerio.
Please use an individual function and align the logic carefully.

Reply:
`we cannot suffice a partitioned requirements`
the father should enfoce an exchange as needed.

others have changed.

Reply:
updated.